### DISCUSSION TITLE
Create networkit graphs via import from a neo4j graph database

### DISCUSSION THREAD

**Discussion by rjurney** [networkit/networkit#1201] [Ideas]
I would like a method to create a networkit Graph from a neo4j database. I would like it to preserve the properties, or at least store them in a data structure to be used with the graph.

**Comment by fabratu** [comment]
Hi!

So far, we don't have any neo4j converter. Also, our experience with the tool is limited. A converter would likely need to know, how to handle different tables from the database and what to do with the relations. We support in C++ arbitrary node and edge attributes (in Python: int, float, string). Maybe a minimal converter would be feasible, or we can guide you for the NetworKit part of the necessary programming.

Would it be possible to share a minimal code example for a neo4j database in your use-case? This might help in getting an idea how we can support such a converter.

**Comment by rjurney** [comment]
I had forgotten that NetworKit is actually a C++ app. Let me do some homework and return.
